templates/etc/init.d/rpm/td-agent

```
@@ -99,17 +99,17 @@ do_stop() {
 	    if [ $RETVAL -eq 0 ]; then
 		success
 	    else
-		failure
+		failure || true
 	    fi
 	fi
 	echo
-	[ $RETVAL -eq 0 ] && rm -f $PIDFILE && rm -f /var/lock/subsys/$prog
+	[ $RETVAL -eq 0 ] && rm -f $PIDFILE && rm -f <%= File.join(root_path, "var/lock/subsys/$prog") %>
 	return $RETVAL
 }
 
 do_restart() {
 	do_configtest || return $?
-	do_stop
+	do_stop || true
```

@repeatedly 
I changed `do_restart` to ignore any errors in `do_stop` to allow starting process on stale `PIDFILE` scenario. This is because we cannot differentiate the following 2 of stop failures,

1. failed to stop process since the process has already been stopped (or, specified process no longer exists)
2. failed to stop process by unknown error

This is a limitation by current implementation since `/bin/kill` and `killproc` (declared in `/etc/init.d/functions`) return different value. The previous behaviour without `set -e`, `do_restart` was trying to invoke `do_start` even if `do_stop` failed. I believe this would be what the user of this script expects.